### DISCUSSION TITLE
【搭建纯静态网页】利用github actions，服务器不安装nodejs

### DISCUSSION THREAD

**Discussion by jarlin8** [tangly1024/NotionNext#135] [Q&A] [answered]
我用2.9版本的[NotionNext](https://github.com/tangly1024/NotionNext)，在github actions跑代码。
`name: 部署到gh-Pages分支

on:
  push:
    branches:
      - main
jobs:
  deploy:
    name: Deploy to GitHub Pages
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - name: 安装yarn
      # 步骤执行指令
        run: npm install -g yarn
      - name: 安装依赖
        run: yarn install
      - name: 打包编译
        run: yarn run build
      - name: Deploy to GitHub Pages
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          exclude_assets: '.github,node_modules'
          publish_dir: .next`
输出的代码 .next下的所有文件，发现不可以像vuepress和mkdocs一样，把静态文件推送到服务器，利用nginx反代访问。
大佬可以回复吗？

**Comment by txs** [comment]
這個專案我還沒用 nextjs SSG 成靜態上傳過，我目前都用 vercel 來 host.
編成靜態我找時間試試看

**Comment by tangly1024** [comment, marked as answer]
利用[next build](https://nextjs.org/docs/advanced-features/static-html-export)的特性可以导出纯静态界面。
可以参考这个[Issue](https://github.com/tangly1024/NotionNext/issues/120)下的讨论，题主想把网页导出成静态html，并部署在Web3.0环境下。

注意，这么做需要修改代码，可以参考分支[feature_html_export](https://github.com/tangly1024/NotionNext/tree/feature_html_export)中的[改动这几处](https://github.com/tangly1024/NotionNext/pull/121/files)。

**Reply by vinson-lee01** [reply]
Expect support

**Reply by vinson-lee01** [reply]
> 利用[next build](https://nextjs.org/docs/advanced-features/static-html-export)的特性可以导出纯静态界面。 可以参考这个[Issue](https://github.com/tangly1024/NotionNext/issues/120)下的讨论，题主想把网页导出成静态html，并部署在Web3.0环境下。
> 
> 注意，这么做需要修改代码，可以参考分支[feature_html_export](https://github.com/tangly1024/NotionNext/tree/feature_html_export)中的[改动这几处](https://github.com/tangly1024/NotionNext/pull/121/files)。

**Comment by vinson-lee01** [comment]
利用[next build](https://nextjs.org/docs/advanced-features/static-html-export)的特性可以导出纯静态界面。
可以参考这个https://github.com/tangly1024/NotionNext/issues/120下的讨论，题主想把网页导出成静态html，并部署在Web3.0环境下。

注意，这么做需要修改代码，可以参考分支[feature_html_export](https://github.com/tangly1024/NotionNext/tree/feature_html_export)中的[改动这几处](https://github.com/tangly1024/NotionNext/pull/121/files)。
